papergres.go

```diff
@@ -407,20 +407,31 @@ func (db *Database) GenerateInsert(obj interface{}) *Query {
 	return db.Schema("public").GenerateInsert(obj)
 }
 
-// Insert inserts the passed in object
+// Insert inserts the passed in object in DB.
+//
+// NOTE: It does not account for client side generated value for a
+// primary key and expects that the logic for populating value
+// of primary key should reside in the database as sequence.
+//
+// DO NOT use Insert() if you wish to populate a client side generated
+// value in primary key, use InsertWithPK() instead.
 func (s *Schema) Insert(obj interface{}) *Result {
-	sql := insertSQL(obj, s.Name)
-	args := insertArgs(obj)
+	sql := insertSQL(obj, s.Name, false)
+	args := insertArgs(obj, false)
 	return s.Database.Query(sql, args...).Exec()
 }
 
-// GenerateInsert generates the insert query for the given object
-func (s *Schema) GenerateInsert(obj interface{}) *Query {
-	sql := insertSQL(obj, s.Name)
-	args := insertArgs(obj)
-	q := s.Database.Query(sql, args...)
-	q.insert = true
-	return q
+// InsertWithPK performs inserts on objects and persists the Primary key value
```

Comment:
See above regarding comments.

Reply:
Same as above. Line comments are the norm.